Sources/Private/CoreAnimation/Layers/ShapeLayer.swift

```diff
@@ -97,44 +97,49 @@ final class GroupLayer: BaseAnimationLayer {
     // recursively creating more `GroupLayer`s
     try setupGroups(from: group.items, parentGroup: group, context: context)
 
-    let (pathDrawingItemsInGroup, otherItemsInGroup) = nonGroupItems.grouped(by: \.item.drawsCGPath)
-
-    // If all of the path-drawing `ShapeItem`s have keyframes with the same timing information,
-    // we can combine the `[KeyframeGroup<BezierPath>]` (which have to animate in separate layers)
-    // into a single `KeyframeGroup<[BezierPath]>`, which can be combined into a single CGPath animation.
-    //
-    // This is how Groups with multiple path-drawing items are supposed to be rendered,
-    // because combining multiple paths into a single `CGPath` (instead of rendering them in separate layers)
-    // allows `CAShapeLayerFillRule.evenOdd` to be applied if the paths overlap. We just can't do this
-    // in all cases, due to limitations of Core Animation.
-    if
-      pathDrawingItemsInGroup.count > 1,
-      let combinedShapeKeyframes = Keyframes.combinedIfPossible(pathDrawingItemsInGroup.map {
-        ($0.item as? Shape)?.path
-      }),
-      // `Trim`s are currently only applied correctly using individual `ShapeItemLayer`s,
-      // because each path has to be trimmed separately.
-      !otherItemsInGroup.contains(where: { $0.item.type == .trim })
-    {
-      let combinedShape = CombinedShapeItem(
-        shapes: combinedShapeKeyframes,
-        name: group.name)
-
-      let sublayer = try ShapeItemLayer(
-        shape: ShapeItemLayer.Item(item: combinedShape, parentGroup: group),
-        otherItems: otherItemsInGroup,
-        context: context)
-
-      addSublayer(sublayer)
-    }
+    // Create `ShapeItemLayer`s for each subgroup of shapes that should be rendered as a single unit
+    //  - These groups are listed from front-to-back, so we have to add the sublayers in reverse order
+    for shapeRenderGroup in nonGroupItems.shapeRenderGroups.reversed() {
+      // If all of the path-drawing `ShapeItem`s have keyframes with the same timing information,
```

Comment:
Everything in this method was just re-indented to account for this new `for` loop